packages/rpc/src/index.ts

```diff
@@ -62,89 +62,83 @@ export const Code = component('rpc' as Component, {
   Unauthorized: '' as StatusCode,
   Forbidden: '' as StatusCode,
   BadRequest: '' as StatusCode,
-  UnknownMethod: '' as StatusCode<{method: string}>
+  UnknownMethod: '' as StatusCode<{ method: string }>
 })
 
 export interface RequestStream {
-  send: <T extends any[]>(req: Request<T>) => void
+  send: (request: Request<any>) => void
 }
 
 interface Operation {
-  id: number
-  request: () => void // Perform request again
+  id: ReqId
   promise: Promise<any>
-  resolve: <T> (value: T) => void
+  resolve: <T>(value: T) => void
+  reject: (reason?: any) => void
 }
 
 /**
- * Process RPC request and handle responses
+ * Process requests and handle responses.
+ * Also allow to handle non identified results passed from other side.
+ *
+ * Hold operations in progress and allow to retry them if required.
  */
-export class RequestProcessor implements Storage {
-  reqId: number = 0
+export class RequestProcessor {
+  reqIndex: number = 0
   requests = new Map<ReqId, Operation>()
   stream: () => Promise<RequestStream>
-  handler: (tx: Tx) => void
 
-  constructor (stream: () => Promise<RequestStream>, handler: (tx: Tx) => void) {
+  constructor (stream: () => Promise<RequestStream>) {
     this.stream = stream
-    this.handler = handler
   }
 
-  public process<T>(response: Response<T>): void {
+  public process (response: Response<any>): void {
     if (response.id !== undefined) {
       const req = this.requests.get(response.id)
       if (req !== undefined) {
-        req.resolve(response)
+        if (response.error !== undefined) {
+          req.reject(response.error)
+        } else {
+          req.resolve(response.result)
+        }
       }
-    } else {
-      // From server transaction
-      this.handler((response.result as unknown) as Tx)
     }
   }
 
-  public onOpen (): void {
+  /**
+   * Reject all waited pending operations.
+   * @param error - error to reject with.
+   */
+  public reject (error: any): void {
     // We need to reply requests in case they are missed.
     for (const op of this.requests.entries()) {
-      op[1].request()
+      op[1].reject(error)
     }
+    this.requests.clear()
   }
```

Comment:
Why do we need this method?

Reply:
To not hang on promise in UI in case of connection drop.

Reply:
Can you please elaborate on this more?